plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java

```diff
@@ -397,6 +399,28 @@ private static String fullTableName(TableId remoteTableId)
         return format("%s.%s.%s", remoteTableId.getProject(), remoteTableId.getDataset(), remoteTableId.getTable());
     }
 
+    public List<RelationCommentMetadata> listRelationCommentMetadata(ConnectorSession session, BigQueryClient client, String schemaName)
+    {
+        TableResult result = client.executeQuery(session, """
+                SELECT tbls.table_name, options.option_value
+                FROM %1$s.`INFORMATION_SCHEMA`.`TABLES` tbls
+                LEFT JOIN %1$s.`INFORMATION_SCHEMA`.`TABLE_OPTIONS` options
+                ON tbls.table_schema = options.table_schema AND tbls.table_name = options.table_name AND options.option_name = 'description'
+                """.formatted(quote(schemaName)));
+        return result.streamValues()
+                .map(row -> {
+                    Optional<String> comment = row.get(1).isNull() ? Optional.empty() : Optional.of(unquote(row.get(1).getStringValue()));
+                    return new RelationCommentMetadata(new SchemaTableName(schemaName, row.get(0).getStringValue()), false, comment);
+                })
+                .collect(toImmutableList());
+    }
+
+    private static String unquote(String quoted)
+    {
+        return quoted.substring(1, quoted.length() - 1)
+                .replace("\"\"", "\"");
```

Comment:
The option_value is quoted as BigQuery official documentation says: https://cloud.google.com/bigquery/docs/information-schema-table-options#example_2

Special characters are tested in `BaseConnectorTest#testCommentTableSpecialCharacter()`.